src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -311,12 +313,18 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
             != a.end();
     }
 
-    static VehicleType getVehicleTypeFromObject(TabObjectEntry& entry)
+    static std::optional<VehicleType> getVehicleTypeFromObject(TabObjectEntry& entry)
     {
         ObjectManager::freeTemporaryObject();
         ObjectManager::loadTemporaryObject(*entry.object._header);
 
         auto vehicleObj = reinterpret_cast<VehicleObject*>(ObjectManager::getTemporaryObject());
+        if (vehicleObj == nullptr)
+        {
+            Logging::info("Could not load determine vehicle type for object '{}', skipping", entry.object._header->getName());
+            return std::nullopt;
+        }
```

Comment:
Why not use the result of loadTemporaryObject?

Reply:
You mean to bail out early if `loadTemporaryObject` failed? I suppose I could do that, yeah.